torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -525,6 +539,17 @@ void reorderExprsForComputeAt(std::vector<Expr*>& exprs) {
 void LoopNestGenerator::generate(const std::vector<Expr*>& exprs) {
   FusionGuard fg(fusion_);
 
+  // Identify all shared memory TensorViews
+  // Initialize Modified status
+  for (auto v : fusion_->vals()) {
+    if (v->getValType().value() == ValType::TensorView) {
+      TensorView* tv = dynamic_cast<TensorView*>(v);
```

Comment:
we already checked the node type, so instead of dynamic_cast we should say `v->as<TensorView>()`. This can also be inlined directly in the if condition

Reply:
done